### ISSUE TITLE
DBT bugs

### ISSUE THREAD

**Issue by Zidras** [DeadlyBossMods/DBM-Unified#135]
**Describe the bug**

I have compiled some visual bugs that happen with DBT.

**Do you have an error log of what happened?**

NA

**To Reproduce**
See below for each case scenario.

**Screenshots**
1. In Classic animation, AnimateEnlarge does not update its anchor Points to the bars latest position, causing the bars to teleport to their final destination once animation is finished.
(2 vids, one expandUpwards, other not): https://imgur.com/a/codsYVk

2. In Classic animation, MoveToNextPosition was not reimplemented and therefore it is not working.
(2 vids, one with 2021 animation, other current): https://imgur.com/a/NHpR0Q3

3. If Fill up and Huge Bars are disabled, Enlarge will make the small bar reset its progress and maximum value.

![image](https://user-images.githubusercontent.com/10605951/178115159-0bbf363c-ea2b-4a17-882a-5b4d04b3cf4e.png)

https://user-images.githubusercontent.com/10605951/178115143-0188175c-86dd-4b40-b40e-8bc2af3c008f.mp4

**Did you try having DeadlyBossMods as the only enabled addon and everything else (especially something like ElvUI) disabled?**
Yes.

**Which version of DeadlyBossMods are you using?**
9.2.21 alpha (20222/07/01 20:33:14)

**Was it working in a previous version? If yes, which was the last good one?**
Before DBT rewrite, from 2021: https://github.com/DeadlyBossMods/DBM-Retail/commit/0d607c755e28739b5eed264a6e0a3e07004ce78a#diff-379aeb92b9b0fea6bbf3d7326c06cc3ea4142a1ab4ea74a64ba49f0e269a138d

**Additional context**
I have already relayed 1 and 2 to Artemis, but since I am finding more issues with DBT, I thought it'd be best to aggregate them in one place. I will be updating this report if I detect more cases, same as I've been doing with GUI bug report.

**Comment by emmericp** [comment]
yeah, looks like the animations are a bit janky, specifically MoveToNextPosition() is implemented but never called. enlarge animation seems to target the wrong "spot"

**Comment by Zidras** [comment]
For 3. , as mentioned by @MysticalOS , the commit   https://github.com/DeadlyBossMods/DeadlyBossMods/commit/d8b1bc36e5acfa94204ae33ecc2a5beaa6b05741 addressed the issue.

However, it created another one, which I want to run by you whether you consider it a **regression**, or now **intended behaviour**.

Every time the small bar enlarges into huge bar, previous behaviour made the bar progress from initial position. Now it keeps the progress.

### Previous behaviour:
(huge bar starts at the beginning)

https://github.com/user-attachments/assets/5f152099-5366-47d5-af05-2c64ce399ee2

### Current behaviour:
(huge bar starts wherever small bar left off)

https://github.com/user-attachments/assets/dfb1232e-d26e-40bd-ad4c-ec94520786fb

**Comment by MysticalOS** [comment]
i don't think there was any way to fix issues without introducing this behavior.

**Comment by Zidras** [comment]
> i don't think there was any way to fix issues without introducing this behavior.

So I can consider this to now be the desired behavior, and not waste time in trying to fix this?

**Comment by MysticalOS** [comment]
if you think you can fix it without regression. i won't be opposed, but I won't go out of my way to try and fix it myself because original problem was worse

plus I know some users actually prefer the bar to restart, that's the bigwigs behavior. so it'd even be nice if it was an option to "restart large bars" vs continuing them.

**Comment by Zidras** [comment]
> if you think you can fix it without regression. i won't be opposed, but I won't go out of my way to try and fix it myself because original problem was worse
> 
> plus I know some users actually prefer the bar to restart, that's the bigwigs behavior. so it'd even be nice if it was an option to "restart large bars" vs continuing them.

I'll see what I can do. 

As for bw behaviour option, that only makes sense on large bars enabled, not small bars, correct?

**Comment by MysticalOS** [comment]
correct, if not using large bars the bar should never restart

**Comment by Zidras** [comment]
and only on noAnim (Simple) option, I assume. Classic animated option would look janky to suddenly restart bar once it reaches large anchor
